File: Codewars/8kyu/8kyu_altERnaTIng_cAsE_ALTerNAtiNG_CaSe/src/main.py
import unittest

def to_alternating_case(string):
    # return ''.join(ch if not ch.isalpha() else ch.upper() if ch.islower() else ch.lower() for ch in string)
    # return string.swapcase()
    return ''.join(ch.upper() if ch.islower() else ch.lower() for ch in string)
    


class Test(unittest.TestCase):
    def test_1(self):
        self.assertEqual(to_alternating_case("hello world"), "HELLO WORLD")

    def test_2(self):
        self.assertEqual(to_alternating_case("HELLO WORLD"), "hello world")

    def test_3(self):
        self.assertEqual(to_alternating_case("hello WORLD"), "HELLO world")

    def test_4(self):
        self.assertEqual(to_alternating_case("HeLLo WoRLD"), "hEllO wOrld")

    def test_5(self):
        self.assertEqual(to_alternating_case("12345"), "12345")

    def test_6(self):
        self.assertEqual(to_alternating_case("1a2b3c4d5e"), "1A2B3C4D5E")

    def test_7(self):
        self.assertEqual(to_alternating_case("String.prototype.toAlternatingCase"), "sTRING.PROTOTYPE.TOaLTERNATINGcASE")

if __name__ == '__main__':
    unittest.main()

File: Codewars/8kyu/8kyu_altERnaTIng_cAsE_ALTerNAtiNG_CaSe/src/main.js
function mysolution() {
    let origin = this.valueOf();
    let ret = '';

    for(let i=0; i<origin.length; i++) {
        let tmp = origin[i];
        ret += (tmp === tmp.toUpperCase())? tmp.toLowerCase(): tmp.toUpperCase();
    }
    return ret;
}

function othersolution() {
    return this.split('').map(a => a === a.toUpperCase()? a.toLowerCase(): a.toUpperCase()).join('')
}

function othersolution2() {
    return [...this].map(ch => ch === ch.toUpperCase()? ch.toLowerCase(): ch.toUpperCase()).join('');
}

String.prototype.toAlternatingCase = othersolution2

console.log("hello world".toAlternatingCase() == "HELLO WORLD");
console.log("HELLO WORLD".toAlternatingCase() == "hello world");
console.log("hello WORLD".toAlternatingCase() == "HELLO world");
console.log("HeLLo WoRLD".toAlternatingCase() == "hEllO wOrld");
console.log("12345".toAlternatingCase() == "12345");
console.log("1a2b3c4d5e".toAlternatingCase() == "1A2B3C4D5E");
console.log("String.prototype.toAlternatingCase".toAlternatingCase() == "sTRING.PROTOTYPE.TOaLTERNATINGcASE");